### ISSUE TITLE
Use non-pickle save/load method

### ISSUE THREAD

**Issue by xingjian-zhang** [Graph-Learning-Benchmarks/gli#266]
**Is your feature request related to a problem? Please describe.**
The pickle-based method is unsafe when the scipy version varies. See https://stackoverflow.com/a/72632847

**Describe the solution you'd like**
We need a new method to save and load arrays without pickling.

**Describe alternatives you've considered**
Ask the contributors to use a lower version when testing fails due to this issue.

**Additional context**
- [Save Load Scipy Sparse Csr Matrix In Portable Data Format
](https://www.faqcode4u.com/faq/456270/save-load-scipy-sparse-csr-matrix-in-portable-data-format)
- [NumPy: consequences of using 'np.save()' with 'allow_pickle=False'](https://stackoverflow.com/questions/41696360/numpy-consequences-of-using-np-save-with-allow-pickle-false)

**Comment by xingjian-zhang** [comment]
**TL;DR**: 
- Separate the storage of scipy sparse arrays and numpy arrays. 
- Require users to save sparse arrays individually by `scipy.sparse.save_npz()`.
- Add support to sparse array reading in file reader through `scipy.sparse.load_npz()`. 

After some investigation, I conclude the issue is caused by numpy not directly supporting loading scipy sparse matrix. Here is an illustration about what I found:

## Save multiple numpy (dense) arrays
```python
import numpy as np
import scipy as sp
from scipy import sparse

a = np.random.random([10, 10])
b = np.random.random([5, 10])
c = np.random.random([10, 5])
# save without pickle
np.savez_compressed("all_np.npz", a=a, b=b, c=c)
# load with np method (without pickle)
f = np.load("all_np.npz", allow_pickle=False)
print(list(f.keys()))
print(np.array_equal(a, f["a"]))
```
```
['a', 'b', 'c']
True
```
By default, if all saved objects are numpy arrays, they are saved without pickle. That means the user can load them safely whatever the version is.

## Save (one) scipy sparse array
```python
s = sparse.rand(10, 10, density=0.1, format="coo")
sparse.save_npz("sp.npz", s)

# load with scipy method (without pickle)
f_sp = sparse.load_npz("sp.npz")
print(type(f_sp))

# load with np method (without pickle)
f_np = np.load("sp.npz", allow_pickle=False)
print(type(f_np))
print(list(f_np.keys()))
```
```
<class 'scipy.sparse._coo.coo_matrix'>
<class 'numpy.lib.npyio.NpzFile'>
['row', 'col', 'format', 'shape', 'data']
```
scipy provides a method that is based on numpy file API. That is, the information of a sparse array is divided into 5 arrays (`['row', 'col', 'format', 'shape', 'data']`). So, we can use `sparse.load_npz()` to re-construct the sparse array given these arrays.

## Save both
```python
np.savez_compressed("np_sp.npz", a=a, b=b, c=c, s=s)
f = np.load("np_sp.npz", allow_pickle=False)
print(np.array_equal(a, f["a"]))
print(type(f["s"]))
```
```
True
ValueError: Object arrays cannot be loaded when allow_pickle=False
```
In this case, numpy arrays and scipy sparse arrays are saved together. Though numpy arrays can still be saved without pickle, the scipy sparse ones are saved via pickle. In other words, numpy does not know how to handle a scipy sparse array, so it directly calls pickle to save it.
